extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesOverlordUtils.java

```diff
@@ -43,4 +45,10 @@ public static String convertTaskIdToK8sLabel(String taskId)
     return taskId == null ? "" : StringUtils.left(RegExUtils.replaceAll(taskId, K8S_TASK_ID_PATTERN, "")
         .toLowerCase(Locale.ENGLISH), 63);
   }
+
+  public static String convertTaskIdToJobName(String taskId)
```

Comment:
whats the limit on length of job name?

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ (its 63 characters)

Reply:
we do need to since there are some characters in task ids that are not valid in k8s labels/job names